applications/RANSApplication/python_scripts/formulations/utilities.py

```diff
@@ -0,0 +1,244 @@
+    model = origin_modelpart.GetModel()
+    connectivity_preserve_modeler = Kratos.ConnectivityPreserveModeler()
+
+    if not model.HasModelPart(destination_modelpart_name):
+        model_part = model.CreateModelPart(destination_modelpart_name)
+
+        # TODO: Remove this line once the warnings from connectivity preserve modeller is gone, otherwise,
+        #       output will be cluttered with lots of missing variable warnings
+        RansVariableUtilities.CopyNodalSolutionStepVariablesList(
+            origin_modelpart, model_part)
+
+        # TODO: [PeriodicCondition]
+        #       Currently, all the conditions will be replaced with the given new condition. This is an issue
+        #       in the case of periodic cases in mpi, there we have to put PeriodicConditions in the mdpa file,
+        #       where MetisParitioner will use that condition list to properly partition it. Therefore, "original_condition_name"
+        #       is not used in this method at the moment.
+        #       Following is one of the proposals to make PeriodicConditions to work with connectivity_preserve_modeller.
+        # connectivity_preserve_modeler.GenerateModelPart(
+        #     origin_modelpart, model_part, element_name, condition_name,
+        #     original_condition_name + str(domain_size) + "D" + str(domain_size)
+        #     + "N")
+
+        connectivity_preserve_modeler.GenerateModelPart(
+            origin_modelpart, model_part, element_name, condition_name)
+
+    Kratos.Logger.PrintInfo("RANSModelPartFactory",
+                            "Created " + destination_modelpart_name)
+    return model.GetModelPart(destination_modelpart_name)
+
+
+def CreateResidualBasedBlockBuilderAndSolver(linear_solver, is_periodic,
+                                             communicator):
+    if (is_periodic):
+        return periodic_block_builder_and_solver(linear_solver, communicator)
+    else:
+        return block_builder_and_solver(linear_solver, communicator)
+
+
+def CreateResidualBasedNewtonRaphsonStrategy(model_part, scheme,
+                                             convergence_criteria,
+                                             builder_and_solver,
+                                             max_iterations, compute_reactions,
+                                             reform_dofs_at_each_step,
+                                             move_mesh_flag):
+    return newton_raphson_strategy(model_part, scheme,
+                                   convergence_criteria, builder_and_solver,
+                                   max_iterations, compute_reactions,
+                                   reform_dofs_at_each_step, move_mesh_flag)
+
+
+def CreateLinearSolver(solver_settings):
+    return linear_solver_factory.ConstructSolver(solver_settings)
+
+
+def CreateResidualCriteria(variable_convergence_criteria_info_list):
+    return residual_criteria(variable_convergence_criteria_info_list)
```

Comment:
These should be implemented in your base solver to follow the current solvers design. Actually, deriving from the `FluidSolver` would avoid reimplementing these.

Reply:
the problem is, the base solver doesn't know anything about how to do `Strategies`, since only leaf formulations knows which `Strategy` is required to solve which equation. I cannot have a common `_CreateNewtonRaphsonStrategy` method in base `RansFormulation` as well because, there can be one `Strategy` required for some formulations (eg. IncompressiblePotentialFlow`, and there can be more than one startegy required by some (eg. k-omega-sst)

Reply:
I did `RansFormulations` to avoid having lots of python files, and to max-re-use the existing code.

Reply:
additional wrapper methods are now removed